app/helpers/effective_truncate_html_helper.rb

```diff
@@ -28,9 +28,8 @@ def truncate_html(text, max = 200, read_more = '', ellipsis = '...')
           textNode = element.children.first
           textNode.content = truncate(textNode.content, length: (max - length), separator: ' ', omission: ellipsis)
           break # Break out of our while loop, as our ellipsis might invalidate our looping condition
-        else # Unexpected, so just remove the whole thing
-          Rails.logger.info "effective_posts, Unexpected number of last_element_child children"
-          element.remove
```

Comment:
@matt-riemer, this line was intended to remove that element from doc, but didn't for some reason. That caused infinite `while` loop, because amount of text nodes remained more than 1.

I decided to remove last text nodes until there remains only 1, and then it truncated its text.